lib/pure/os.nim

```diff
@@ -338,6 +341,62 @@ proc expandFilename*(filename: string): string {.rtl, extern: "nos$1",
       result = $r
       c_free(cast[pointer](r))
 
+proc normalizePath*(path: var string, separator=DirSep) {.rtl, extern: "nos$1", tags: [].} =
+  ## Normalize a path.
+  ##
+  ## Consecutive directory separators are collapsed, including an initial double slash.
+  ##
+  ## On relative paths, double dot (..) sequences are collapsed if possible.
+  ## On absolute paths they are always collapsed.
+  ##
+  ## Warning: URL-encoded and Unicode attempts at directory traversal are not detected.
+  ## Triple dot is not handled.
+  let is_abs = path[0] == separator
```

Comment:
not a big deal, but convention in stdlib is `camelCase`.

Reply:
Also, on Windows this is wrong, no? You should use the ``isAbsolute`` procedure.